cmd/agent/app/agent.go

```diff
@@ -65,13 +69,26 @@ func NewAgentCommand(ctx context.Context) *cobra.Command {
 		},
 	}
 
-	// Init log flags
-	// TODO(@RainbowMango): Group the flags to "logs" flag set.
-	klog.InitFlags(flag.CommandLine)
+	fss := cliflag.NamedFlagSets{}
+
+	genericFlagSet := fss.FlagSet("generic")
+	// Add the flag(--kubeconfig) that is added by controller-runtime
+	// (https://github.com/kubernetes-sigs/controller-runtime/blob/v0.11.1/pkg/client/config/config.go#L39),
+	// and update the flag usage.
+	genericFlagSet.AddGoFlagSet(flag.CommandLine)
+	genericFlagSet.Lookup("kubeconfig").Usage = "Path to karmada control plane kubeconfig file."
```

Comment:
For `karmada-agent`, no need to update the `Usage`. This is different with `karmada-controller-manager`.
For `karmada-agent`
- `--karmada-kubeconfig` --> point to karmada control plane.
- `--kubeconfig` --> point to the cluster where running `karmada-agent`

Remove this line and the comments on above should be ok.